hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/webserver/UIWebServer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2016-2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.diff.ui.webserver;
+
+
+import com.netflix.hollow.diff.ui.HollowDiffUI;
+import com.netflix.hollow.diff.ui.HollowDiffUIRouter;
+import com.netflix.hollow.diff.ui.webserver.HollowDiffUIServer.UIServer;
+import com.netflix.hollow.tools.diff.HollowDiff;
+import com.sun.net.httpserver.HttpServer;
+import java.net.InetSocketAddress;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+final class UIWebServer implements UIServer {
+    private HttpServer server;
+    private final HttpHandlerWithServletSupport handler;
+    private final HollowDiffUIRouter router;
+    private final int port;
+    private ExecutorService threadPool;
+
+    public UIWebServer(int port) {
+        this.port = port;
+        this.router = new HollowDiffUIRouter();
+        this.handler = new HttpHandlerWithServletSupport(this.router);
+        this.threadPool = Executors.newCachedThreadPool();
+    }
+
+    public void start() throws Exception {
+        server = HttpServer.create(new InetSocketAddress(port), 0);
+        server.createContext("/", this.handler);
+        server.setExecutor(threadPool);
+        server.start();
+    }
+
+    public void join() throws InterruptedException {
+        while (!threadPool.isShutdown()) {
+            Thread.sleep(10);  // sleep for a few ms
+        }
+    }
+
+    public void stop() throws Exception {
+        threadPool.shutdown();
```

Comment:
shtudown doesnt cancel perviously submitted tasks that might currently be executing. Look into the two phase shutdown described here https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html

Reply:
done